### DISCUSSION TITLE
`onReset` callback for Form component

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#45044] [Ideas]
Hey, it would be nice to trigger custom logic when the form is reset (with resetFileds) using an onReset callback prop.

**Comment by zombieJ** [comment]
`onReset` is still native event for the Form component. But I think it could be HOC of it:

```tsx
const useMyForm = (resetCallback) => {
  const [form] = Form.useForm;

  return React.useMemo(() => {
    // Maybe proxy is better
    return {
      ...form,
      resetFields: (...args) => {
        resetCallback?.();
        form.resetFields(...args);
      },
    };
  }, [form]);
};
```

**Reply by bombillazo** [reply]
Thanks for the suggestion.

I would prefer it be a prop on Form and not a parameter of myForm setup hook.

I am trying to experiment with my own custom form component. I am creating a new `useForm` hook but I am having trouble syncing the form instance returned in this hook with the actual form logic that is handled by my custom Form component. How does native AntD allow for the Form.useForm hook to return a form instance and then the Form component can pass props like onFinish to the form?

**Reply by bombillazo** [reply]
I ended up making a patch over rc-field-form which works like a charm! Ill share it here if case it helps others with the missing form features: https://github.com/react-component/field-form/pull/620
